packages/collection-model/index.d.ts

```diff
@@ -40,18 +51,10 @@ interface Collection {
     fetchInfo?: boolean;
     force?: boolean;
   }): Promise<void>;
-  fetchMetadata(opts: { dataService: DataService }): Promise<{
-    namespace: string;
-    isReadonly: boolean;
-    isTimeSeries: boolean;
-    isClustered: boolean;
-    isFLE: boolean;
-    isSearchIndexesSupported: boolean;
-    sourceName?: string;
-    sourceReadonly?: boolean;
-    sourceViewon?: string;
-    sourcePipeline?: unknown[];
-  }>;
+  fetchMetadata(opts: {
+    dataService: DataService;
```

Comment:
Should we type this as well and pick only the methods needed in the model? This way it would make it easy when doing something [like this](https://github.com/mongodb-js/compass/blob/f903236b2151ff6573b761271075dd4247978ce6/packages/compass-saved-aggregations-queries/src/stores/open-item.ts#L228) in a plugin where we have DataService typed for the corresponding plugin.

Reply:
That seems to be the last place that calls this method outside the instance store loop, let me remove that while I'm at it

Reply:
Removed this direct usage in https://github.com/mongodb-js/compass/pull/4858/commits/b58813da58faa19431b37f8b60f322d232ee7f19